extensions/notebook/src/test/protocol/notebookUriHandler.test.ts

```diff
@@ -107,9 +118,50 @@ describe('Notebook URI Handler', function (): void {
 				.reply(httpErrorCode);
 
 			await notebookUriHandler.handleUri(notebookUri);
-			
 			sinon.assert.callCount(showErrorMessageSpy, 1);
 			sinon.assert.notCalled(showNotebookDocumentStub);
 		});
 	});
+
+	it('should open notebook when file uri is valid', async function (): Promise<void> {
+		let showQuickPickStub = sinon.stub(vscode.window, 'showQuickPick').resolves(Promise.resolve(loc.msgYes) as any);
+		let notebookDir: string = path.join(os.tmpdir(), `notebook_${uuid.v4()}`);
+		let notebookPath: string = path.join(notebookDir, 'hello.ipynb');
+
+		try {
+			// Remove file if it exists
+			await promisify(rimraf)(notebookDir);
+			await fs.mkdir(notebookDir);
+			let fileURI = 'azuredatastudio://microsoft.notebook/open?url=file://' + notebookPath;
+			let fileNotebookUri = vscode.Uri.parse(fileURI);
+			let notebookContent: azdata.nb.INotebookContents = {
+				cells: [{
+					cell_type: CellTypes.Code,
+					source: ['x = 1 \ny = 2'],
+					metadata: { language: 'python', tags: ['parameters'] },
+					execution_count: 1
+				}],
+				metadata: {
+					kernelspec: {
+						name: 'python3',
+						language: 'python',
+						display_name: 'Python 3'
+					}
+				},
+				nbformat: 4,
+				nbformat_minor: 5
+			};
+
+			await fs.writeFile(notebookPath, JSON.stringify(notebookContent));
+
+			await notebookUriHandler.handleUri(fileNotebookUri);
+
+			sinon.assert.calledOnce(showQuickPickStub);
+			sinon.assert.calledWith(showNotebookDocumentStub, sinon.match.any, sinon.match({ initialContent: notebookContent }));
+			sinon.assert.callCount(showErrorMessageSpy, 0);
+
+		} catch (error) {
+			// No-op
```

Comment:
Well now if an actual error happens during the test you'll never know.

Since you're now generating a UUID for the folder I don't think it's necessary to delete the folder at all - and thus can also remove this try/catch.

Reply:
Will utilize UUID by itself now.